src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1041,16 +1050,29 @@ impl<'a, 'tcx> DocVisitor for LinkCollector<'a, 'tcx> {
         // Rather than merging all documentation into one, resolve it one attribute at a time
         // so we know which module it came from.
         for (parent_module, doc) in item.attrs.collapsed_doc_value_by_module_level() {
+            if !may_have_doc_links(&doc) {
+                continue;
```

Comment:
What is the goal of the short-circuiting here? Rustdoc has already loaded the attributes from disk. All this avoids is a call to `markdown_links`, but that function is both not very expensive and much more reliable than this check.

Reply:
It is expensive, https://github.com/rust-lang/rust/pull/96135#issuecomment-1101475868.